src/Microsoft.VisualStudio.Threading.Analyzers.Tests/Helpers/CodeFixVerifier.cs

```diff
@@ -1,186 +0,0 @@
-
-            // We'll go through enough for each diagnostic to be caught once
-            bool fixApplied = false;
-            for (int i = 0; i < attempts && analyzerDiagnostics.Length > 0; ++i)
-            {
-                var diagnostic = analyzerDiagnostics[0]; // just get the first one -- the list gets smaller with each loop.
-                var document = project.GetDocument(diagnostic.Location.SourceTree);
-                var actions = new List<CodeAction>();
-                var context = new CodeFixContext(document, diagnostic, (a, d) => actions.Add(a), CancellationToken.None);
-                codeFixProvider.RegisterCodeFixesAsync(context).Wait();
-                if (!actions.Any())
-                {
-                    break;
-                }
-
-                var action = codeFixChooser != null ? actions.Single(codeFixChooser) : actions.Single();
-                document = ApplyFix(document, action);
-                fixApplied = true;
-                project = document.Project;
-
-                this.logger.WriteLine("Code after fix:\n{0}", document.GetSyntaxRootAsync().Result.ToFullString());
-
-                analyzerDiagnostics = GetSortedDiagnosticsFromDocuments(ImmutableArray.Create(analyzer), project.Documents.ToArray());
-                var newCompilerDiagnostics = GetNewDiagnostics(
-                    compilerDiagnostics,
-                    project.Documents.SelectMany(doc => GetCompilerDiagnostics(doc)));
-
-                // Check if applying the code fix introduced any new compiler diagnostics
-                if (!allowNewCompilerDiagnostics && newCompilerDiagnostics.Any())
-                {
-                    // Format and get the compiler diagnostics again so that the locations make sense in the output
-                    document = document.WithSyntaxRoot(Formatter.Format(document.GetSyntaxRootAsync().Result, Formatter.Annotation, document.Project.Solution.Workspace));
-                    newCompilerDiagnostics = GetNewDiagnostics(compilerDiagnostics, GetCompilerDiagnostics(document));
-
-                    Assert.True(false,
-                        string.Format("Fix introduced new compiler diagnostics:\r\n{0}\r\n\r\nNew document:\r\n{1}\r\n",
-                            string.Join("\r\n", newCompilerDiagnostics.Select(d => d.ToString())),
-                            document.GetSyntaxRootAsync().Result.ToFullString()));
-                }
-            }
-
-            if (newSources != null && newSources[0] != null)
-            {
-                Assert.True(fixApplied, "No code fix offered.");
-
-                // After applying all of the code fixes, compare the resulting string to the inputted one
-                int j = 0;
-                foreach (var document in project.Documents)
-                {
-                    var actual = GetStringFromDocument(document);
-                    Assert.Equal(newSources[j++], actual, ignoreLineEndingDifferences: true);
```

Comment:
> ignoreLineEndingDifferences: true

:memo: The new test library does not make this provision, but I moved it to #366

Reply:
Thanks for reviewing equivalence of what you're removing with what you're replacing it with.